docs/api/toolkits.md

```diff
@@ -0,0 +1,112 @@
+
+```python
+>>> from openff.toolkit.utils.toolkits import GLOBAL_TOOLKIT_REGISTRY as toolkit_registry
+>>> from openff.toolkit import Molecule
+>>> molecule = Molecule.from_smiles('Cc1ccccc1')
+>>> smiles = toolkit_registry.call('to_smiles', molecule)
+
+```
+
+The order of toolkits, as specified in `toolkit_precedence` above, determines the order in which
+the called method is resolved, i.e. if the toolkit with highest precedence has a `to_smiles`
+method, that is the toolkit that will be called. If the toolkit with highest precedence does not
+have such a method, it is attempted with other toolkits until one is found. By default, if a toolkit with an appropriately-named method raises an exception of any type, then iteration over the registered toolkits stops and that exception is raised. To continue iteration if specific exceptions are encountered, customize this behavior using the optional `raise_exception_types` keyword argument to `ToolkitRegistry.call`. If no registered
+toolkits have the method, a ValueError is raised, containing a message listing the registered toolkits and exceptions (if any) that were ignored. 
+
+Alternatively, the global toolkit registry (which will attempt to register any available toolkits) can be used:
+
+```python
+>>> from openff.toolkit.utils.toolkits import GLOBAL_TOOLKIT_REGISTRY as toolkit_registry
+>>> len(toolkit_registry.registered_toolkits)
+4
+
+```
+
+Individual toolkits can be registered or deregistered to control the backend that ToolkitRegistry calls resolve to. This can
+be useful for debugging and exploring subtley different behavior between toolkit wrappers.
+
+To temporarily change the state of `GLOBAL_TOOLKIT_REGISTRY`, we provide the `toolkit_registry_manager`
+context manager.
+
+```python
+>>> from openff.toolkit.utils.toolkits import RDKitToolkitWrapper, AmberToolsToolkitWrapper, GLOBAL_TOOLKIT_REGISTRY
+>>> from openff.toolkit.utils import toolkit_registry_manager
+>>> print(len(GLOBAL_TOOLKIT_REGISTRY.registered_toolkits))
+4
+>>> with toolkit_registry_manager(ToolkitRegistry([RDKitToolkitWrapper(), AmberToolsToolkitWrapper()])):
+...     print(len(GLOBAL_TOOLKIT_REGISTRY.registered_toolkits))
+2
+
+```
+
+To remove `ToolkitWrappers` permanently from a `ToolkitRegistry`, the `deregister_toolkit` method can be used:
+
+```python
+>>> from openff.toolkit.utils.toolkits import OpenEyeToolkitWrapper, BuiltInToolkitWrapper
+>>> from openff.toolkit.utils.toolkits import GLOBAL_TOOLKIT_REGISTRY as toolkit_registry
+>>> print(len(toolkit_registry.registered_toolkits))
+4
+>>> toolkit_registry.deregister_toolkit(RDKitToolkitWrapper)
+>>> print(len(toolkit_registry.registered_toolkits))
+3
+>>> toolkit_registry.register_toolkit(RDKitToolkitWrapper)
+>>> print(len(toolkit_registry.registered_toolkits))
+4
+
+```
+
+For example, differences in `to_smiles` functionality between OpenEye toolkits and The RDKit can
+be explored by selecting which toolkit(s) are and are not registered.
```

Comment:
If a user want to do this, I'd suggest they use the context manager, not this approach. I see this as a useful example for demonstrating the mechanics of the registry, but could we tack on at the end a sort of "by the way, here's another way to do the same thing" section? I think it would look like

```python
>>> # imports
>>> molecule = Molecule.from_smiles('Cc1ccccc1')
>>> with toolkit_registry_manager(ToolkitRegistry([OpenEyeToolkitWrapper()]):
...     print(molecule.to_smiles())
[H]c1c(c(c(c(c1[H])[H])C([H])([H])[H])[H])[H]
>>> with toolkit_registry_manager(ToolkitRegistry([RDKitToolkitWrapper()]):
...     print(molecule.to_smiles())
[H][c]1[c]([H])[c]([H])[c]([C]([H])([H])[H])[c]([H])[c]1[H]
```

I may be biased here - I am lazy and I like being able to get this in one go with the context manager.

Reply:
Interesting - Is there a good reason I'm not familiar with not to just use the `toolkit_registry` argument?

Reply:
For one-liners it's just style

For larger sequences of logic (imagine setting up conformers and later calling out to some cornucopia of `openmmforcefields` and NAGL), it's still a question of style but less ambiguous


```
with toolkit_registry_manager(no_openeye_registry):
    # do thing
    # do other thing
    # oh no there's more
    # oh god it doesn't stop
    # but wait there's more
```

and I'd rather not wire that through on each line